base-local/docker-compose.yaml.fs

```diff
@@ -1,103 +1,97 @@
-          - "./files/users.conf:/etc/sftp/users.conf:ro"
-          - "rawfiles:/home"
+services:
+  database:
+    build: ./database
+    image: database
+    ports:
+      - "${READIUM_DATABASE_PORT}:3306"
+    restart: always
+    environment:
+      MYSQL_ROOT_PASSWORD: "${READIUM_DATABASE_PASSWORD}"
+    volumes:
+      - "dbdata:/var/lib/mysql"
 
-    lcpencrypt:
-        image: "readium/lcpencrypt:working"
-        build:
-            context: .
-            target: "lcpencrypt"
-        environment:
-          - "READIUM_LCPSERVER_HOST=http://192.168.1.253:8989"
-          - "READIUM_LCPSERVER_USERNAME=lcp"
-          - "READIUM_LCPSERVER_PASSWORD=secretpassword"
-        volumes:
-          - "encfiles:/opt/readium/files/encrypted"
-          - "rawfiles:/opt/readium/files/raw"
+  sftp:
+    image: "atmoz/sftp:alpine"
+    ports:
+      - "${READIUM_SFTP_PORT}:22"
+    volumes:
+      - "./files/users.conf:/etc/sftp/users.conf:ro"
+      - "rawfiles:/home"
 
-    lcpserver:
-        image: "readium/lcpserver:working"
-        build:
-            context: .
-            target: "lcpserver"
-        environment:
-          - "READIUM_DATABASE_HOST=192.168.1.253"
-          - "READIUM_DATABASE_PORT=3326"
-          - "READIUM_DATABASE_USERNAME=root"
-          - "READIUM_DATABASE_PASSWORD=secretpassword"
-          - "READIUM_DATABASE_DATABASE=lcpserver"
-          - "READIUM_LCPSERVER_HOST=http://192.168.1.253:8989"
-          - "READIUM_LSDSERVER_HOST=http://192.168.1.253:8990"
-          - "READIUM_FRONTEND_HOST=http://192.168.1.253:8991"
-          - "READIUM_ENC_CONTENT=/opt/readium/files/encrypted"
-        ports:
-          - 8989:8989
-        volumes:
-          - "encfiles:/opt/readium/files/encrypted"
-        depends_on:
-          - database
+  lcpencrypt:
+    <<: *default-app
+    image: "readium/lcpencrypt:working"
+    build:
+      context: .
+      target: "lcpencrypt"
+    volumes:
+      - "encfiles:/opt/readium/files/encrypted"
+      - "rawfiles:/opt/readium/files/raw"
 
-    lsdserver:
-        image: "readium/lsdserver:working"
-        build:
-            context: .
-            target: "lsdserver"
-        environment:
-          - "READIUM_DATABASE_HOST=192.168.1.253"
-          - "READIUM_DATABASE_PORT=3326"
-          - "READIUM_DATABASE_USERNAME=root"
-          - "READIUM_DATABASE_PASSWORD=secretpassword"
-          - "READIUM_DATABASE_DATABASE=lsdserver"
-          - "READIUM_LCPSERVER_HOST=http://192.168.1.253:8989"
-          - "READIUM_LSDSERVER_HOST=http://192.168.1.253:8990"
-          - "READIUM_FRONTEND_HOST=http://192.168.1.253:8991"
-        ports:
-          - 8990:8990
-        depends_on:
-          - database
+  lcpserver:
+    <<: *default-app
+    image: "readium/lcpserver:working"
+    build:
+      context: .
+      target: "lcpserver"
+    ports:
+      - "${READIUM_LCPSERVER_PORT}:8989"
+    volumes:
+      - "encfiles:/opt/readium/files/encrypted"
+      - "./etc:/etc/readium"
```

Comment:
I know this was to get a copy of the certificate into the container. But the certificate should (will, I think) already be in the container based on the build instructions. I'll try a run without this bind mount and see. On the other hand, if we want to use this bind mount (to keep a sensitive certificate out of the image), we should also investigate to see if we can remove line 32 (original) of the Dockerfile.

Reply:
I think we can keep both of them. It will allow us to get a default certificate in the image initially and then later override  if required